WordPress/gutenberg
# Remove empty column blocks from mobile in editor, to match front-end

## What?
Fixes https://github.com/WordPress/gutenberg/issues/63933 by not always rendering an empty column block when viewing the mobile preview in the editor.  

## Why?
The front-end and editor instances of the Columns block should be exactly the same. That way, I can design in the editor and it renders on the front-end exactly like I intended. 

## Testing Instructions

1. Open a post or page.
2. Add a columns block with two columns. You can use [this block markup](https://gist.github.com/richtabor/5f52c1e8d2610b275bc1fd988671f718) if you'd like. I colorized the columns to make them clear. 
3. Switch to mobile
4. See no blank space where the empty column is. 
5. Check that the front-end mobile view matches. 
6. Go back to the editor. 
7. Select the parent Columns block. 
8. Check you can insert blocks into the empty column when the parent columns block is selected or the block itself is selected. 

## Screenshots or screencast 

https://github.com/user-attachments/assets/2b7ffca5-5307-4cb5-ad74-b6910d95e639
